### PR TITLE
Add NetworkManager to make WiFi configuration easier.

### PR DESCRIPTION
**Pull request by prfi**
This pull request adds support for Network Manager on RPi3 platform. Network manager should simplify managing WiFi connections. This is first step to adding graphical interface to managing WiFi.

### CODE REVIEW

**Comment by bukepo** [comment]
To keep consistent with existing scripts, could you please use spaces instead of tabs to indent?

**Comment by prfi** [comment]
Done

File: script/_dhcpv6_pd [hunk 1]
```diff
@@ -126,9 +135,20 @@ WATCH1="type='signal', interface=\${WPANTUND_IFACE}, path=\${WPANTUND_NCP_STATE_
 
 reload_dhcpcd_daemon()
 {
-        systemctl force-reload dhcpcd
-        systemctl reload-daemon
-        logger -t "\${NAME}[\${PID}]" "Reload: dhcpcd daemon"
+    if systemctl is-active NetworkManager; then
+            logger -t "\${NAME}[\${PID}]" "NetworkManager: active"
```
Comment (bukepo) [inline script/_dhcpv6_pd]: indent by 4 spaces?

File: script/_dhcpv6_pd [hunk 2]
```diff
@@ -183,7 +203,15 @@ dhcpv6_pd_install()
     if have systemctl; then
         create_dhcpcd_reloader_service
 
+    sudo systemctl daemon-reload
```
Comment (bukepo) [inline script/_dhcpv6_pd]: indent?

File: script/_network_manager [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+    ;;
+esac
+
+EOF
+}
+
+create_dhcpv6_helper_script()
+{
+    sudo tee ${DHCPV6_HELPER_SCRIPT} <<EOF
+#!/bin/bash
```
Comment (bukepo) [inline script/_network_manager]: use `#!/bin/sh`?

File: script/_network_manager [hunk 4]
```diff
@@ -0,0 +1,397 @@
...
+    IFNAME=$(nmcli d | grep ethernet | cut -d" " -f1)
+
+    sudo nmcli c add type ethernet ifname ${IFNAME} con-name ${ETH_CONN}
+    sudo nmcli c modify ${ETH_CONN} ipv6.method ignore
+}
+
+create_ap_helper_script()
+{
+    sudo tee ${AP_HELPER_SCRIPT} <<EOF
+#!/bin/bash
```
Comment (bukepo) [inline script/_network_manager]: use `#!/bin/sh`?

File: script/_dhcpv6_pd [hunk 5]
```diff
@@ -126,18 +140,29 @@ WATCH1="type='signal', interface=\${WPANTUND_IFACE}, path=\${WPANTUND_NCP_STATE_
...
 reload_dhcpcd_daemon()
 {
-        systemctl force-reload dhcpcd
-        systemctl reload-daemon
-        logger -t "\${NAME}[\${PID}]" "Reload: dhcpcd daemon"
+    if systemctl is-active NetworkManager; then
+        logger -t "\${NAME}[\${PID}]" "NetworkManager: active"
+
+        for interface in \$(pgrep dhcpcd -a -x | rev | cut -d" " -f1 | rev); do
+                logger -t "\${NAME}[\${PID}]" "dhcpcd: rebind \${interface}"
```
Comment (bukepo) [inline script/_dhcpv6_pd]: nit: indent.

File: script/_dhcpv6_pd [hunk 6]
```diff
@@ -126,18 +140,29 @@ WATCH1="type='signal', interface=\${WPANTUND_IFACE}, path=\${WPANTUND_NCP_STATE_
...
-        while read -r line; do
-                if echo \${line} | grep -q "associated"; then
-                        reload_dhcpcd_daemon
-                fi
-        done
+    while read -r line; do
+        if echo \${line} | grep -q "associated"; then
+            reload_dhcpcd_daemon
+        fi
+   done
```
Comment (bukepo) [inline script/_dhcpv6_pd]: nit: indent

File: script/_dns64 [hunk 7]
```diff
@@ -51,7 +51,7 @@ dns64_update_resolvconf()
     if [ $PLATFORM = ubuntu ]; then
         sudo resolvconf -u || true
     elif [ $PLATFORM = raspbian ]; then
-        sudo systemctl restart dhcpcd || true
+    sudo systemctl restart NetworkManager || true
```
Comment (bukepo) [inline script/_dns64]: Shall we check if NETWORK_MANAGER is enabled? The `with/without` commands are designed to allow overridden by environment variables.
nit: indent.

**Pull request merged**